src/client/lazy-app/util/index.ts

```diff
@@ -216,13 +237,27 @@ export function drawableToImageData(
   return ctx.getImageData(0, 0, width, height);
 }
 
-export async function builtinDecode(blob: Blob): Promise<ImageData> {
+export async function builtinDecode(
+  signal: AbortSignal,
+  blob: Blob,
+  mimeType: string,
+): Promise<ImageData> {
+  // If WebCodecs are supported, use that.
+  if (await WebCodecs.isTypeSupported(mimeType)) {
+    if (signal.aborted) return Promise.reject('Aborted');
+    try {
+      return await WebCodecs.decode(blob, mimeType);
+    } catch (e) {}
+  }
+  if (signal.aborted) return Promise.reject('Aborted');
+
   // Prefer createImageBitmap as it's the off-thread option for Firefox.
   const drawable =
     'createImageBitmap' in self
       ? await createImageBitmap(blob)
       : await blobToImg(blob);
 
+  if (signal.aborted) return Promise.reject('Aborted');
```

Comment:
```suggestion

  // Prefer createImageBitmap as it's the off-thread option for Firefox.
  const drawable = await abortable(
    signal,
    'createImageBitmap' in self ? createImageBitmap(blob) : blobToImg(blob),
  );
```

Reply:
I applied the same thinking here: Couldn’t there be a scenario where `createImageBitmap` or `blobToImg` start doing work even though we already got an abort signal?

Reply:
Also updated